tests/integration/test_backups.py

```diff
@@ -304,13 +294,7 @@ async def test_restore_on_new_cluster(ops_test: OpsTest, github_secrets) -> None
 
     # Wait for the restore to complete.
     async with ops_test.fast_forward():
-        await wait_for_idle_on_blocked(
-            ops_test,
-            database_app_name,
-            0,
-            S3_INTEGRATOR_APP_NAME,
-            ANOTHER_CLUSTER_REPOSITORY_ERROR_MESSAGE,
-        )
```

Comment:
I'd expect that the `ANOTHER_CLUSTER_REPOSITORY_ERROR_MESSAGE` blocked status would still be set if we tried to restore a backup to a different cluster. The new cluster could have a timeline with an equal number to the new timeline in the original cluster if we also perform a restore there.

Reply:
I've double checked this:
First of all, it should be safe to allow the new restored cluster to work with s3 stanza as after restore this cluster chooses the new timeline based on the stanza settings. For this particular example, the new restored cluster `new-postgresql-k8s` will operate on the timeline 4 as it supposed to be after 2 restores in the `test_backup_gcp` and the third one in the `test_restore_on_new_cluster` (1 -> 2 -> 3 -> 4).
Secondly, this blocked status emerges for the restored cluster due to the `stanza.get("name") != self.charm.app_peer_data.get("stanza", self.stanza_name)` check in the `can_use_s3_repository` func. Basically, it ensures the name of the juju model + application must be the same.
Because of the second point, I've restored this piece of code (this change was uncorrect at the first place) and now this test is correct.